server/routerlicious/packages/routerlicious-base/src/alfred/app.ts

```diff
@@ -76,26 +65,18 @@ export function create(
     app.use(compression());
     const loggerFormat = config.get("logger:morganFormat");
     if (loggerFormat === "json") {
-        app.use(morgan((tokens, req, res) => {
-            const messageMetaData = {
-                method: tokens.method(req, res),
-                pathCategory: `${req.baseUrl}${req.route ? req.route.path : "PATH_UNAVAILABLE"}`,
-                url: tokens.url(req, res),
-                driverVersion: tokens.req(req, res, DriverVersionHeaderName),
-                status: tokens.status(req, res),
-                contentLength: tokens.res(req, res, "content-length"),
-                durationInMs: tokens["response-time"](req, res),
-                tenantId: getTenantIdFromRequest(req.params),
-                documentId: getIdFromRequest(req.params),
-                serviceName: "alfred",
-                eventName: "http_requests",
-             };
-             logRequestMetric(messageMetaData);
-             winston.info("request log generated", { messageMetaData });
-             return undefined;
-        }, { stream }));
+        app.use(
+            jsonMorganLoggerMiddleware(
+                "alfred",
+                (tokens, req, res) => {
+                    return {
+                        [HttpProperties.driverVersion]: tokens.req(req, res, DriverVersionHeaderName),
+                        [BaseTelemetryProperties.tenantId]: getTenantIdFromRequest(req.params),
```

Comment:
qq. Why do we want to add tenantid and doc id as a part of additional properties and not in default set of properties in jsonMorganLoggerMiddleware itself if it exists? If we pass this responsibility on to the individual callers, isn't there a greater chance of it not being included inadvertently?

Reply:
Because the way we calculate tenantId, documentId is different for different callers. For example:
1. Riddler does not compute documentId, while Alfred does
2. GitRest computes TenantId and DocumentId from the Storage-Routing-Id: https://github.com/microsoft/FluidFramework/blob/c78794f0462043bf77b8b270c4e17c8561be858e/server/gitrest/packages/gitrest-base/src/app.ts#L43
3. Historian computes documentId based on the authorization token/header: https://github.com/microsoft/FluidFramework/blob/8bbe5a015b5d3e4b5601b3bc6cfd3785ead8d48c/server/historian/packages/historian-base/src/app.ts#L67